test/e2e/webview.test.ts

```diff
@@ -0,0 +1,30 @@
+import { promises as fs } from "fs"
+import * as path from "path"
+import { describe, test, expect } from "./baseFixture"
+
+describe("Webviews", ["--disable-workspace-trust"], {}, () => {
+  test("should preview a Markdown file", async ({ codeServerPage }) => {
+    // Create Markdown file
+    const heading = "Hello world"
+    const dir = await codeServerPage.workspaceDir
+    const file = path.join(dir, "text.md")
+    await fs.writeFile(file, `# ${heading}`)
+    await codeServerPage.openFile(file)
+
+    // Open Preview
+    await codeServerPage.executeCommandViaMenus("Markdown: Open Preview to the Side")
+    // Wait for the iframe to open and load
+    await codeServerPage.page.waitForTimeout(2000)
+    await codeServerPage.waitForTab(`Preview ${file}`)
+
+    let totalCount = 0
+    for (const frame of codeServerPage.page.frames()) {
+      // Check for heading in frames
+      const count = await frame.locator(`text=${heading}`).count()
+      totalCount += count
+    }
+
+    // One in the file and one in the preview
+    expect(totalCount).toBe(2)
```

Comment:
I was having trouble figuring out how to look for elements within the `iframe`. In the doc examples, they expect the iframe to have an `id` which you can use to locate the frame.

One thing surprising to me is when I inspect the DOM inside code-server, I see the `src` for the `iframe` is the `pre/index.html` but when i log the `frame.url` in this for loop, I see two frames and one is the `webWorkerExtensionHostIframe.html` so a little confused. I guess `url` and `src` are not the same?

Anyways, this test locates the heading text and counts the selectors then adds it to the total. Total should be two since in file and one in the preview. I don't love this approach so I'm open to other ideas

Reply:
I bet there are at least some class names we could use.

Reply:
Ahh that makes sense. Okay, I'll try the `frameLocator` and see if that works. Thanks man!